crates/e2e-testing/src/spin_controller.rs

```diff
@@ -46,6 +49,22 @@ impl Controller for SpinUp {
     ) -> Result<AppInstance> {
         let appdir = spin::appdir(app_name);
 
+        //cleanup state dir before spin up
+        let state_dir = match env::var("STATE_DIR") {
```

Comment:
Arguably, if someone is using a custom state directory, then it is on them to manage it; I agree with @lann that it could be dangerous to delete anything the tests don't "own"? Maybe simply (unconditionally) delete the `.spin` directory if it exists?